addon/mixins/scoped-query-cache.js

```diff
@@ -73,5 +76,21 @@ export default Ember.Mixin.create({
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    this.decayId = setTimeout(() => {
+      this.cache = this._createCache();
+
+      callback && typeof callback === 'function' && callback();
```

Comment:
Here is an example

```
import Ember from 'ember';
import ScopedQueryCacheMixin from 'ember-scoped-query-cache/mixins/scoped-query-cache';

export default Ember.Route.extend({
  deactivate() {
    setTimeout(() => {
      this.queryCache.clearCache();
    }, 10000);
  }
});
```

With this approach, not only do we take the complexity out of the library, we also give the user full flexibility